manifests/profile.pp

```diff
@@ -0,0 +1,51 @@
+# Updates a ``dconf`` profile entry to ``/etc/dconf/profile/$name``
+#
+# @see man 7 dconf
+#
+# @param name
+#   The name of the profile file in ``/etc/dconf/profile``
+#
+# @param entries
+#   One or entries in the following Hash format:
+#
+#   @example Profile Hierarchy Hash
+#     'user':          # Name of the database
+#       'type': 'user' # DB Type
+#       'order': 0     # Priority order (optional, defaults to 15)
+#
+#   * The suggested default hierarchy used by the module data is as follows:
+#     * User DB   => 0
+#     * System DB => Between 11 and 39
+#
+# @param target
+#   The target directory within which to create the profile
+#
+define dconf::profile (
+  Dconf::DBSettings    $entries,
+  Stdlib::AbsolutePath $target   = '/etc/dconf/profile'
+) {
+
+  include 'dconf'
+
+  ensure_resource('file', $target, {
```

Comment:
I woudln't think it would be a big deal to require a class where this common stuff can be put to be included before this define could be used.

We probably shouldn't be adding too much code that uses ensure_resource anyway